lib/liquid/variable_lookup.rb

```diff
@@ -15,8 +14,8 @@ def initialize(markup)
       lookups = markup.scan(VariableParser)
 
       name = lookups.shift
-      if name =~ SQUARE_BRACKETED
-        name = Expression.parse(Regexp.last_match(1))
+      if name&.start_with?('[') && name&.end_with?(']')
```

Comment:
I suspect the perf gain is only due to this (and bellow). This looks like a nice optimization to me.